plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveBucketFunction.java

```diff
@@ -31,29 +38,85 @@
     private final BucketingVersion bucketingVersion;
     private final int bucketCount;
     private final List<TypeInfo> typeInfos;
+    private final List<Type> partitionColumnTypes;
```

Comment:
This is effectively making `HiveBucketFunction` a union type which is more complicated and harder to understand. Instead, let's create a separate implementation for partitions.